### PR TITLE
Added controllers to Authentification, Movie and MovieSession links

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 @ComponentScan(basePackages = {
         "com.dev.cinema.dao",
         "com.dev.cinema.service",
+        "com.dev.cinema.mapper",
```
Comment (DariiaPikul) [inline src/main/java/com/dev/cinema/config/AppConfig.java:18]: Doesn't it work without this line?

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    public AuthenticationController(AuthenticationService service, UserMapper mapper) {
+        this.authenticationService = service;
+        this.userMapper = mapper;
+    }
+
+    @SneakyThrows
+    @PostMapping("/registration")
+    public void doUserRegister(@RequestBody UserRequestDto userRequestDto) {
+        User user = userMapper.dtoToUser(userRequestDto);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: Do we need this variable?

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        movieSessionService.add(movieSessionMapper.dtoToMovieSession(requestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailable(@RequestParam Long movieId,
+                                                         @RequestParam String date) {
+        List<MovieSession> movieSessions =
+                movieSessionService.findAvailableSessions(movieId,
+                        LocalDate.parse(date, DateTimeFormatter.ofPattern(DATE_FORMAT)));
+
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: Do we need a separate variable and this empty line?

File: src/main/java/com/dev/cinema/mapper/MovieMapper.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.dev.cinema.dto.movie.MovieRequestDto;
+import com.dev.cinema.dto.movie.MovieResponseDto;
+import com.dev.cinema.model.Movie;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieMapper {
+
+    public Movie dtoToMovie(MovieRequestDto movieRequestDto) {
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/mapper/MovieMapper.java]:
```suggestion
public class MovieMapper {
    public Movie dtoToMovie(MovieRequestDto movieRequestDto) {
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: > Do we need this variable?

let's remove it

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+public class MovieController {
+    private final MovieService movieService;
+    private final MovieMapper movieMapper;
+
+    public MovieController(MovieService movieService, MovieMapper movieMapper) {
+        this.movieService = movieService;
+        this.movieMapper = movieMapper;
+    }
+
+    @PostMapping("/")
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]:
```suggestion
    @PostMapping
```

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    private static final String DATE_FORMAT = "d.MM.yyyy";
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService service, MovieSessionMapper mapper) {
+        this.movieSessionService = service;
+        this.movieSessionMapper = mapper;
+    }
+
+    @PostMapping("/")
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: same

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    public void addMovieSession(@RequestBody MovieSessionRequestDto requestDto) {
+        movieSessionService.add(movieSessionMapper.dtoToMovieSession(requestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailable(@RequestParam Long movieId,
+                                                         @RequestParam String date) {
+        List<MovieSession> movieSessions =
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: let's use single stream here

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @PostMapping("/")
+    public void addMovieSession(@RequestBody MovieSessionRequestDto requestDto) {
+        movieSessionService.add(movieSessionMapper.dtoToMovieSession(requestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailable(@RequestParam Long movieId,
+                                                         @RequestParam String date) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java:36]: you can consider this approach 
@RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date

File: src/main/java/com/dev/cinema/mapper/UserMapper.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package com.dev.cinema.mapper;
+
+import com.dev.cinema.dto.user.UserRequestDto;
+import com.dev.cinema.dto.user.UserResponseDto;
+import com.dev.cinema.model.User;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/mapper/UserMapper.java]: do we need this mapper?

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: removed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 7 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 8 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java:36]: fixed

File: src/main/java/com/dev/cinema/mapper/MovieMapper.java (hunk 4 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/mapper/MovieMapper.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 5 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/MovieController.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 6 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: fixed

File: src/main/java/com/dev/cinema/mapper/UserMapper.java (hunk 9 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/mapper/UserMapper.java]: removed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**